### ISSUE TITLE
IDE Bug: "Verify Code After Upload" preference does not work

### ISSUE THREAD

**Issue by cmaglie** [arduino/Arduino#1344]
Relevant topic:

http://arduino.cc/forum/index.php/topic,154141.msg1156478.html

Original text:

```
I assume we post bugs here... if not, someone please point me to the right place!

Regardless of the state of the "Verify Code After Upload" checkbox
in Preferences, the code sent to the Due is always verified.  A little bit 
of troubleshooting on my part revealed that the culprit is the '-v' flag 
that is always passed to 'bossac':

bossac.exe --port=COM30 -U false -e -w -v -b [temp_filename] -R

If this command is run manually without the -v flag, verification does 
not occur.

This bug occurs on Windows in both the beta and the nightly build,
I can't vouch for Linux or MacOS.  The verification takes several 
seconds to run even on a small sketch so it's nice to be able to turn it off.
```

**Comment by mathewjury** [comment]
This issue is still current in 1.6.4. The above recommendation by ergodic00 didn;t help. Compiling for a Nano board if thats of any assistance.

Mat.

**Comment by hallard** [comment]
Hi there, I confirm this bug is still present, but I don't thing it's related to -v flag (which is verbose), in fact in the IDE when this option is unchecked the switch -V (upper case) should be added to command line, I just tested by hand and it works.
Pay attention, I 1st put this -V at the end of the command line and it did not work, then I tried putting it just after avrdude it worked, see below

```
$ avrdude -V -CS:/blablabla/avrdude.conf -patmega328p -cusbtiny -Uflash:w:C:/Users/CHARLES/AppData/Local/Temp/build8905307267913391739.tmp/bootloader_test.cpp.hex:i
avrdude.exe: AVR device initialized and ready to accept instructions
Reading | ################################################## | 100% -0.00s
avrdude.exe: Device signature = 0x1e950f
avrdude.exe: NOTE: "flash" memory has been specified, an erase cycle will be performed
             To disable this feature, specify the -D option.
avrdude.exe: erasing chip
avrdude.exe: reading input file "C:/Users/CHARLES/AppData/Local/Temp/build8905307267913391739.tmp/bootloader_test.cpp.hex"
avrdude.exe: writing flash (13596 bytes):
Writing | ################################################## | 100% 3.34s
avrdude.exe: 13596 bytes of flash written
avrdude.exe done.  Thank you.
```

**Comment by gh-megabit** [comment]
Just to add (it didn't seem clear here) - this affects uploads to many boards using avrdude, not just bossa.

I can't find the feature implemented in the code at all, except for the preferences form.

I'm working around this with a manual change to the platforms.txt file.

**Comment by EGNBR** [comment]
with this "solution" the preference flag is istill ignored, but it elinate the verification pass.

# AVR Uploader/Programmers tools
# ------------------------------

tools.avrdude.path={runtime.tools.avrdude.path}
tools.avrdude.cmd.path={path}/bin/avrdude
tools.avrdude.config.path={path}/etc/avrdude.conf

tools.avrdude.upload.params.verbose=-v
tools.avrdude.upload.params.quiet=-q -q
###### modified line, "-V" added after "config" ###########
tools.avrdude.upload.pattern="{cmd.path}" "-C{config.path}" -V {upload.verbose} -p{build.mcu} -c{upload.protocol} -P{serial.port} -b{upload.speed} -D "-Uflash:w:{build.path}/{build.project_name}.hex:i"

**Comment by mathewjury** [comment]
Great! These instructions worked well. Thanks EGNBR!

**Comment by cimba007** [comment]
Why is this still not fixed?

**Comment by DamonHD** [comment]
Any idea when this will be fixed?  Verify takes us an extra minute each time on our current set-up and essentially never fails.

**Comment by gh-megabit** [comment]
I've submitted a pull request for this - hoping that it is accepted

**Comment by steronydh** [comment]
Bump. This still doesn't work in 1.6.9 nightly and I have to edit platforms.txt to prevent the verification pass.

**Comment by gh-megabit** [comment]
New pull request created to fix this = #4873

**Comment by cmaglie** [comment]
fixed by #4873
